pkg/cloudprovider/types.go

```diff
@@ -90,8 +91,18 @@ type InstanceType struct {
 	Overhead *InstanceTypeOverhead
 }
 
+// allocatableCache caches the allocatable resources for each instance type.  It makes the assumption that once an
+// instance type reports allocatable resources, it won't change in the future. This is important for correctness. This
+// change improved the scheduling performance ~20-30% in testing.
+var allocatableCache sync.Map
```

Comment:
IIUC, instance types vary depending on NodeTemplate, so you can get name collision.

Reply:
Modified to cache in the instance type itself.